src/main/java/jp/co/yahoo/yosegi/hadoop/tools/logsampler/LogSampler.java

```diff
@@ -0,0 +1,219 @@
+    if ( input == null || input.isEmpty() ) {
+      throw new IOException( "input is null or empty." );
+    }
+    FileSystem fs = FileSystem.get( getConf() );
+    Path inputPath = new Path( input );
+
+    if ( ! fs.exists( inputPath ) ) {
+      throw new IOException( "input path is not exists. " + input );
+    }
+
+    if ( ! fs.isDirectory( inputPath ) ) {
+      throw new IOException( "input path is not directory. " + input );
+    }
+    return inputPath;
+  }
+
+  /**
+   * Create output Path object.
+   */
+  public Path createOutputDirectoryPath( final String output ) throws IOException {
+    if ( output == null || output.isEmpty() ) {
+      throw new IOException( "output is null or empty." );
+    }
+    FileSystem fs = FileSystem.get( getConf() );
+    Path outputPath = new Path( output );
+
+    if ( fs.exists( outputPath ) ) {
+      throw new IOException( "output path is not exists. " + output );
+    }
+    return outputPath;
+  }
+
+  /**
+   * Get spread count.
+   */
+  public long getSpreadCount( final Path inputPath )
+      throws IOException, InterruptedException, ClassNotFoundException {
+    Configuration conf = getConf();
+    Job job = Job.getInstance( conf, "Yosegi get spread count." );
+    job.setMapperClass( GetStatsMapper.class );
+    job.setNumReduceTasks(0);
+    job.setOutputKeyClass( NullWritable.class );
+    job.setOutputValueClass( NullWritable.class );
+    job.setInputFormatClass( YosegiStatsInputFormat.class );
+    YosegiStatsInputFormat.addInputPath( job , inputPath );
+    job.setOutputFormatClass( NullOutputFormat.class );
+    if ( ! job.waitForCompletion(true) ) {
+      throw new IOException( "Failed get spread count." );
+    }
+    return job.getCounters().findCounter( "yosegi_stats" , "spreads" ).getValue();
+  }
+
+  /**
+   * Create sample file.
+   */
+  public void createSampleFile(
+      final Path inputPath ,
+      final Path outputPath ,
+      final long sampleCount ,
+      final long spreadCount ) throws IOException, InterruptedException, ClassNotFoundException {
+    Configuration conf = getConf();
+    conf.setLong( "yosegi.tools.logsampler.samples" , sampleCount );
+    conf.setLong( "yosegi.tools.logsampler.spreads" , spreadCount );
+    Job job = Job.getInstance( conf, "Yosegi create sample file." );
+    job.setMapperClass( LogSamplerMapper.class );
+    job.setReducerClass( LogSamplerReducer.class );
+    job.setNumReduceTasks(1);
+    job.setInputFormatClass( YosegiSpreadInputFormat.class );
+    YosegiSpreadInputFormat.addInputPath( job , inputPath );
+    job.setOutputFormatClass( YosegiParserOutputFormat.class );
+    YosegiParserOutputFormat.setOutputPath( job, outputPath );
+
+    if ( ! job.waitForCompletion(true) ) {
+      throw new IOException( "Failed create sample file." );
+    }
+  }
+
+  /**
+   * Create input file list.
+   */
+  public static void printHelp( final String[] args ) {
+    HelpFormatter hf = new HelpFormatter();
```

Comment:
Java Doc comment seems not link to the function of the method.
It might be print help or some from the method name.

Reply:
Thank you.
I corrected the comment.

Reply:
Thank you.